turbinia/api/cli/README.md

```diff
@@ -1 +1,164 @@
+       %%%                         %%  %%             %%%           %%%%,
+       %%%      %%%   %%%   %%%%%  %%%   %%%   %%  %%%   %%%  %%%       (%%
+       %%%      %%%   %%%  %%%     %%     %%/  %%  %%%   %%%  %%%  %%%%%%%%
+       %%%      %%%   %%%  %%%     %%%   %%%   %%  %%%   %%%  %%% %%%   %%%
+       %%%        %%%%%    %%%       %%%%%     %%  %%%    %%  %%%   %%%%%
+
+  This command-line tool interacts with Turbinia's API server.
+
+  You can specify the API server location in ~/.turbinia_api_config.json
+
+Options:
+  -c, --config_instance TEXT  A Turbinia instance configuration name.
+                              [default: (dynamic)]
+  -p, --config_path TEXT      Path to the .turbinia_api_config.json file..
+                              [default: (dynamic)]
+  -h, --help                  Show this message and exit.
+
+Commands:
+  config  Get Turbinia configuration.
+  jobs    Get a list of enabled Turbinia jobs.
+  result  Get Turbinia task or request results.
+  status  Get Turbinia request/task status.
+  submit  Submit new requests to the Turbinia API server.
+  ```
+
+### Getting the server configuration
+```
+turbinia-client config list
+```
+
+### Getting request or task information
+To get the status of a Turbinia request:
+```
+turbinia-client status request <request_id>
+```
+
+For task status:
+```
+turbinia-client status task <task_id>
+```
+where <request_id> and <task_id> are the respective Turbinia request or task identifiers.
+
+To get a summary of all existing requests:
+```
+turbinia-client status summary
+```
+
+### Creating new requests
+New Turbinia requests can be submitted via turbinia-client using the ```submit``` command. In it's simplest form, you only need to pass the evidence type and any required arguments for the specific evidence type. As an example, to submit a new Turbinia request to process a ```RawDisk`` evidence type, run the following command:
+```
+turbinia-client submit rawdisk --source_path /evidence/rawdisk.dd
+```
+Each evidence type will have its own set of required and optional arguments. You can view all possible arguments with:
+```
+turbinia-client submit <evidence_type> -h
+```
+where <evidence_type> is a valid Turbinia evidence name.
+
+A list of all valid Turbinia evidence names can be viewed with:
+```
+turbinia-client submit -h
+```
+
+### Creating custom requests
+It is possible to customze requests with additional parameters. For example, you can provide your own Turbinia recipe and add it to a new Turbinia request as follows:
+```
+turbinia-client submit rawdisk --source_path /evidence/rawdisk.dd --recipe_name /home/user/my_triage_recipe.yaml --requester my_user --reason forensics_case_12345
+```
+
+An alternative way of providing a Turbinia recipe is to use the ```recipe_data``` argument. The ```recipe_data``` argument takes in a string value. It expects the string to be the Base64 encoded content of a valid Turbinia recipe.
```

Comment:
*takes in a Base64 encoded string value of a valid Turbinia recipe reads a little better